lib/transactions.ts

```diff
@@ -62,20 +62,27 @@ export class TransactionSender {
       throw new Error("No transactions passed");
     }
 
+    if (opts?.CUs) {
+      if ((!Array.isArray(opts.CUs) && Array.isArray(txs.length)) || (Array.isArray(opts.CUs) && opts.CUs.length !== txs.length)) {
+        throw new Error("CUs length must match transactions length");
+      }
+    }
+
     const sequence =
       txs[0] instanceof Array ? (txs as T[][]) : ([txs] as T[][]);
 
     const latestBlockhash = await connection.getLatestBlockhash();
     const timedTxs = sequence.map((set) =>
-      set.map((e: T) => {
+      set.map((e: T, i:number) => {
         const tx = e;
         if (!(tx instanceof VersionedTransaction)) {
           tx.recentBlockhash = latestBlockhash.blockhash;
           tx.feePayer = this.owner!;
           // Compute limit ix & priority fee ix
+          const units = Array.isArray(opts?.CUs) ? opts.CUs[i] as number : opts?.CUs;
```

Comment:
So I'd much prefer that we send the transaction what we think we need vs having things calculated within the function. It just adds complexity for something that is complex. If you want to use the transaction sender we should be doing some heavy lifting for it... 

If that doesn't make sense, let me know. But I think I'd rather have this thing dead simple to reason about and have the layer above it more complex than the other way around.

Reply:
This solution is the simple one because we'd need the max CUs for initialize proposals for it to pass and didn't want to set that on the stone. If you don't pass CUs we just put 200_000

Didn't want to take time for that but tell me if I need to make it more advanced

Reply:
Sorry what I meant with this was:

1. The constructor of the transaction should know it's compute
2. Aggregating the constructed transactions should then take the compute and sum across
3. Then that sum should be passed to the transaction sender (so the sender isn't in the business of calculating compute, it just inserts it in.